hal/src/rtl872x/i2c_hal.cpp

```diff
@@ -150,14 +152,21 @@ class I2cClass {
             i2cInitStruct_.I2CMaster  = I2C_MASTER_MODE;
             i2cInitStruct_.I2CAckAddr = 0; // This is the peer device's slave address in master mode, not needed herein.
         } else {
+            if (hal_interrupt_is_isr()) {
+                SPARK_ASSERT(false);
+            }
+            slaveRxCache_ = std::make_unique<uint8_t[]>(slaveRxCacheDepth_);
+            if (!slaveRxCache_) {
+                SPARK_ASSERT(false);
+            }
             i2cInitStruct_.I2CMaster  = I2C_SLAVE_MODE;
             i2cInitStruct_.I2CAckAddr = address; // This is the local device's slave address in slave mode
         }
         I2C_Init(i2cDev_, &i2cInitStruct_);
 
         // Enable restart function
         i2cDev_->IC_CON |= BIT_CTRL_IC_CON_IC_RESTART_EN;
-        i2cDev_->IC_RX_TL = std::min(rxBuffer_.size(), (size_t)255); // Maximum RX FIFO depth: 255 + 1
+        i2cDev_->IC_RX_TL = std::min(rxBuffer_.size(), (size_t)0); // FIFO depth 1
```

Comment:
Isnt this always going to be `0`? 

I think its ok to have a low (or even zero) RX threshold, but we should make sure that is the intent here. 
Note: the documentation for this register says the values can range from 0-255, "with the additional restriction that it may not be set to a value larger than the depth of the buffer". The "buffer" they are referring to is the 16 byte buffer implemented in hardware.

If we do change this to not be zero, it should not be more than 16.

Reply:
I know the FIFO can range from 0-255, but it won't work reliably if we set the depth to a certain value. As what I observed, if for example we set it to 16, the ISR won't trigger even if we have received 16 bytes and sometimes the SCL will be stretched and the master finally timeout before ISR is triggered.

Reply:
Does setting `i2cDev_->IC_RX_TL=0` ensure that the ISR will always be triggered on each byte reception? 
In other words, have you seen the ISR not trigger when only receiving one byte?

Reply:
Not really. When thee first byte is received and ISR is triggeered, we need to continuously read available data as long as the RX FIFO is not empty in the ISR, i.e. the slave might have received several more bytes when we execute the ISR. If we only read one byte in ISR, we might have lost data. It seams that the slave hardware will continue accepting data and set flags accordingly, regardless of whether software has handled this flags in time. In the section 19.2.8.1.3 of the UM0400 (user manual), it emphasize something specifically.

Reply:
@scott-brust No need to change this for master as well, we are not using interrupt mode and not preconfiguring the number of bytes to be recvd, see `requestFrom()` implementation where we just manually trigger receiving each byte with `IC_DATA_CMD` in a loop, so this setting does not affect master behavior at all. All good.